includes/class.llms.frontend.forms.php

```diff
@@ -104,6 +110,7 @@ public function mark_complete() {
 	 * @param  int $lesson_id [ID of the current lesson]
 	 *
 	 * @return void
+	 * @version 3.2.4
 	 */
 	function mark_course_complete( $user_id, $lesson_id ) {
```

Comment:
this function should be removed in favor of LLMS_Student::is_complete() check

logic to mark complete should also be moved to the LLMS_Student class

new function similar to LLMS_Student::enroll() should be created, in pseudo code:

```
function mark_complete( $object_id, $trigger = '' ) {

   // check to be sure post can be completed (only courses, lessons, sections, and tracks can be completed at this moment)

   if term_exists() to check it it is a track
      $obj = get_term
  elseif  get_post() && in_array( post_type, [course, lesson, section] )
      $obj = get the instance of the LLMS_Post_Instance
      // you can cherry-pick the llms_get_post() function out of the scaffolding branch at https://github.com/gocodebox/lifterlms/blob/scaffolding/includes/llms.functions.core.php#L613 (and add section to it)
  else 
     return false

    if $this->is_complete()
       $this->insert_complete_postmeta()

      // cascade up (check "parent" for completion, this way if you complete a lesson, you don't have to check a sections, course, track everytime

      if lesson && $this->is_complete( lesson->get_section() ) {
           $this->mark_complete( lesson->get_section() )
      if section && $this->is_complete( section->get_course() ) {
            $this->mark_complete( section->get_course() )
      if etc...

       return true;
   else
       return false;

}
```